core/src/clients/github.rs

```diff
@@ -263,4 +268,73 @@ impl GraphQLClient {
             Err(e) => Err(anyhow!("Error getting merge queue: {}", e)),
         }
     }
+
+    #[instrument(skip(self))]
+    pub async fn get_commit_statuses(
+        &self,
+        owner: &str,
+        repo: &str,
+        limit: i64,
+    ) -> Result<CommitStatusMap> {
+        let mut map = HashMap::new();
+        match post_graphql::<GetCommitStatuses, _>(
+            &self.client,
+            GITHUB_GRAPHQL_URL,
+            get_commit_statuses::Variables {
+                owner: owner.to_string(),
+                name: repo.to_string(),
+                limit,
+            },
+        )
+        .await
+        {
+            Ok(res) => {
+                let target: GetCommitStatusesRepositoryDefaultBranchRefTarget = res
+                    .data
+                    .ok_or(anyhow!("Failed to get valid response data"))?
+                    .repository
+                    .ok_or(anyhow!("Failed to get valid repository"))?
+                    .default_branch_ref
+                    .ok_or(anyhow!("Failed to get valid default branch ref"))?
+                    .target
+                    .ok_or(anyhow!("Failed to get valid target"))?;
+
+                match target {
+                    GetCommitStatusesRepositoryDefaultBranchRefTarget::Commit(commit) => {
+                        commit
+                            .history
+                            .nodes
+                            .ok_or(anyhow!("Failed to get valid nodes"))?
+                            .into_iter()
+                            .for_each(|node| {
+                                if let Some(node) = node {
+                                    let oid = node.oid.clone();
+                                    if let Some(status) = node.status {
+                                        let short_oid = oid.chars().take(8).collect::<String>();
+                                        let status_str: &str = match status.state {
+                                            get_commit_statuses::StatusState::SUCCESS => "success",
+                                            get_commit_statuses::StatusState::FAILURE => "failure",
+                                            get_commit_statuses::StatusState::ERROR => "error",
+                                            get_commit_statuses::StatusState::EXPECTED => {
+                                                "expected"
+                                            }
```

Comment:
weird that this one is formatted differently than the others :P

Reply:
it's longer :)